contracts/token/ERC20/extensions/ERC20Votes.sol

```diff
@@ -24,267 +22,31 @@ import "../../../utils/cryptography/ECDSA.sol";
  *
  * _Available since v4.2._
  */
-abstract contract ERC20Votes is ERC20Permit, IERC5805 {
-    struct Checkpoint {
-        uint32 fromBlock;
-        uint224 votes;
-    }
-
-    bytes32 private constant _DELEGATION_TYPEHASH =
-        keccak256("Delegation(address delegatee,uint256 nonce,uint256 expiry)");
-
-    mapping(address => address) private _delegates;
-    mapping(address => Checkpoint[]) private _checkpoints;
-    Checkpoint[] private _totalSupplyCheckpoints;
-
-    /**
-     * @dev Clock used for flagging checkpoints. Can be overridden to implement timestamp based checkpoints (and voting).
-     */
-    function clock() public view virtual override returns (uint48) {
-        return SafeCast.toUint48(block.number);
-    }
-
-    /**
-     * @dev Description of the clock
-     */
-    // solhint-disable-next-line func-name-mixedcase
-    function CLOCK_MODE() public view virtual override returns (string memory) {
-        // Check that the clock was not modified
-        require(clock() == block.number, "ERC20Votes: broken clock mode");
-        return "mode=blocknumber&from=default";
-    }
-
-    /**
-     * @dev Get the `pos`-th checkpoint for `account`.
-     */
-    function checkpoints(address account, uint32 pos) public view virtual returns (Checkpoint memory) {
```

Comment:
This function was present in `next-v5.0` prior to the recent merges.

Reply:
The conflict that cause its removal is the following:

- `Checkpoint` is now available as a member of `Checkpoint.History`
- `Checkpoint.History` is (unofficially) deprecated in favor of Trace224, because the "vote with timestamp" needs to control the values used for checkpoints (history forces blockNumber)
- In Votes we use Trace224, so in an ERC20Votes that uses Votes, the type is not available

Here is a commit that re implements that, but that returns a Checkpoint224 (its the same layout, but with different fields name): https://github.com/OpenZeppelin/openzeppelin-contracts/commit/73cf027231de996c8583c002a50dd1ad7b782af7

Note that both `checkpoints(address,uint32)` and `numCheckpoints(address)` are leftover from Compound, are not available in ERC721Votes, and are AFAIK only used for testing. I'd like to discuss removing them in 5.0